### PR TITLE
[CHONKER] Creates Contrib SDK & Abstracts some hook functions

### PR DESCRIPTION
**Pull request by jzeiders**

### CODE REVIEW

**Comment by jzeiders** [comment]
It's the refactoring CHONKER! It just keeps growing

File: api/hackerLive.js [hunk 1]
```diff
@@ -4,149 +4,117 @@ const express = require("express");
...
+  });
+});
 
-    return res.json({ contribs, person });
-  } catch (e) {
-    return res.status(400).json({ err: e.message });
-  }
+router.get("/personInfo", async (req, res) => {
+  const person = await models.Person.findByPk(req.user.id);
+  return res.json({ success: person });
```
Comment (saresend) [inline api/hackerLive.js]: Does this also provide all the tasks and stuff?

File: api/live.js [hunk 2]
```diff
@@ -96,153 +92,134 @@ async function handleContrib(userId, req, res) {
...
+  });
+  if (!isCreated) {
+    return res.status(400).json({
+      error: "User already has completed this task"
+    });
   }
+  return res.json({
+    success: {
+      contribution: result,
+      message: `Successfully created a task contribution for ${profile.firstName} ${profile.lastName}`
```
Comment (saresend) [inline api/live.js]: Is this encoded as a type in and of itself?

File: api/live.js [hunk 3]
```diff
@@ -96,153 +92,134 @@ async function handleContrib(userId, req, res) {
...
-              required: false
-            }
-          ]
-        }
-      ]
-    });
-    const multipliers = result.get("Grouping").get("Multipliers");
-    return multipliers.reduce((x, y) => x + y.dataValues.multiplierValue, 0);
-  } catch (e) {
-    Sentry.captureMessage(e.message);
```
Comment (saresend) [inline api/live.js]: Wait don't delete the try catch here, this uses failures to assume the default multiplier value, and doesn't actually represent a potential user failure.

File: lib/api-sdk/hook-utils.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+export function useErrorHandler(handler = console.log, error?) {
+  useUpdateEffect(() => {
+    if (error && error != "") handler(error), [error];
+  });
+}
+
+// LOOK AT THIS HIGHER ORDER FUNCTION
+export function fetchWithMutation<T extends (...args: any[]) => any>(
```
Comment (saresend) [inline lib/api-sdk/hook-utils.tsx]: Big flex chill

**Pull request merged**